packages/analytics-js/src/app/RudderAnalytics.ts

```diff
@@ -129,29 +128,25 @@ class RudderAnalytics implements IRudderAnalytics<IAnalytics> {
   }
 
   /**
-   * Get preloaded events in buffer queue if exists
-   */
-  // eslint-disable-next-line class-methods-use-this
-  getPreloadBuffer() {
-    const preloadedEventsArray = Array.isArray((globalThis as typeof window).rudderanalytics)
-      ? ((globalThis as typeof window).rudderanalytics as PreloadedEventCall[])
-      : ([] as PreloadedEventCall[]);
-
-    // Expose buffer to global objects
-    setExposedGlobal(GLOBAL_PRELOAD_BUFFER, clone(preloadedEventsArray));
-  }
-
-  /**
-   * Trigger load event in buffer queue if exists
+   * Trigger load event in buffer queue if exists and stores the
+   * remaining preloaded events array in global object
    */
   triggerBufferedLoadEvent() {
     const preloadedEventsArray = Array.isArray((globalThis as typeof window).rudderanalytics)
       ? ((globalThis as typeof window).rudderanalytics as PreloadedEventCall[])
       : ([] as PreloadedEventCall[]);
 
+    // The array will be mutated in the below method
+    promotePreloadedConsentEventsToTop(preloadedEventsArray);
```

Comment:
what happens if multible consents preloaded call happens? the last one will become the first and then the order will get wrong?

Reply:
taken care off